vm/systemSmartContracts/eei.go

```diff
@@ -0,0 +1,201 @@
+
+	selfDestruct map[string][]byte
+}
+
+// NewVMContext creates a context where smart contracts can run and write
+func NewVMContext(blockChainHook vmcommon.BlockchainHook, cryptoHook vmcommon.CryptoHook) (*vmContext, error) {
+	if blockChainHook == nil {
+		return nil, vm.ErrNilBlockchainHook
+	}
+	if cryptoHook == nil {
+		return nil, vm.ErrNilCryptoHook
+	}
+
+	vmc := &vmContext{blockChainHook: blockChainHook, cryptoHook: cryptoHook}
+	vmc.CleanCache()
+
+	return vmc, nil
+}
+
+// SelfDestruct destroy the current smart contract, setting the beneficiary for the liberated storage fees
+func (host *vmContext) SelfDestruct(beneficiary []byte) {
+	host.selfDestruct[string(host.scAddress)] = beneficiary
+}
+
+// GetStorage get the values saved for a certain key
+func (host *vmContext) GetStorage(key []byte) []byte {
+	strAdr := string(host.scAddress)
+	if _, ok := host.storageUpdate[strAdr]; ok {
+		if value, ok := host.storageUpdate[strAdr][string(key)]; ok {
+			return value
+		}
+	}
+
+	data, err := host.blockChainHook.GetStorageData(host.scAddress, key)
+	if err != nil {
+		return nil
+	}
+
+	return data
+}
+
+// SetStorage saves the key value storage under the address
+func (host *vmContext) SetStorage(key []byte, value []byte) {
+	strAdr := string(host.scAddress)
+
+	if _, ok := host.storageUpdate[strAdr]; !ok {
+		host.storageUpdate[strAdr] = make(map[string][]byte, 0)
+	}
+
+	length := len(value)
+	host.storageUpdate[strAdr][string(key)] = make([]byte, length)
+	copy(host.storageUpdate[strAdr][string(key)][:length], value[:length])
+}
+
+// GetBalance returns the balance of the given address
+func (host *vmContext) GetBalance(addr []byte) *big.Int {
+	strAdr := string(addr)
+	if outAcc, ok := host.outputAccounts[strAdr]; ok {
+		actualBalance := big.NewInt(0).Add(outAcc.Balance, outAcc.BalanceDelta)
+		return actualBalance
+	}
+
+	balance, err := host.blockChainHook.GetBalance(addr)
+	if err != nil {
+		return nil
+	}
+
+	host.outputAccounts[strAdr] = &vmcommon.OutputAccount{
+		Balance:      big.NewInt(0).Set(balance),
+		BalanceDelta: big.NewInt(0),
+		Address:      addr}
+
+	return balance
+}
+
+// Transfer handles any necessary value transfer required and takes
+// the necessary steps to create accounts
+func (host *vmContext) Transfer(
+	destination []byte,
+	sender []byte,
+	value *big.Int,
+	input []byte,
+) error {
+
+	senderAcc, ok := host.outputAccounts[string(sender)]
+	if !ok {
+		senderAcc = &vmcommon.OutputAccount{
+			Address:      sender,
+			BalanceDelta: big.NewInt(0),
+			Balance:      big.NewInt(0),
+		}
+		host.outputAccounts[string(senderAcc.Address)] = senderAcc
+	}
+
+	destAcc, ok := host.outputAccounts[string(destination)]
+	if !ok {
+		destAcc = &vmcommon.OutputAccount{
+			Address:      destination,
+			BalanceDelta: big.NewInt(0),
+			Balance:      big.NewInt(0),
+		}
+		host.outputAccounts[string(destAcc.Address)] = destAcc
+	}
+
+	senderAcc.BalanceDelta = big.NewInt(0).Sub(senderAcc.BalanceDelta, value)
```

Comment:
You can call 
```GO
senderAcc.BalanceDelta.Sub(senderAcc.BalanceDelta, value)
```
Valid also for the next line

Reply:
linter will say that output is not used :|.

Reply:
changed.